src/types/Token.ts

```diff
@@ -1,10 +1,19 @@
-interface Token {
+type CustomTokenPart<T> = {
+  [P in keyof T]: T[P]
+};
+
+type StaticTokenPart = {
   access_token: string;
   expires_in?: number;
   scope?: string[];
   token_type?: string;
   local_expiry?: number;
-  [key: string]: any;
 }
 
-export { Token };
+type Token<T = any> = CustomTokenPart<T> & StaticTokenPart;
```

Comment:
Cool 👍 Would you add this change to the PR?

Regarding the default type: I would prefer it as is, do not see a reason to force the user to pass a type, furthermore it would mean another breaking change.